test/framework/clients.go

```diff
@@ -45,15 +45,16 @@ func NewClients(configPath, clusterName, namespace string) *Clients {
 	var err error
 	c := &Clients{}
 
-	c.KubeClient, err = knativetest.NewKubeClient(configPath, clusterName)
+	cfg, err := knativetest.BuildClientConfig(configPath, clusterName)
 	if err != nil {
-		log.Fatalf("failed to create kubeclient from config file at %s: %s", configPath, err)
+		log.Fatalf("failed to create configuration obj from %s for cluster %s: %s", configPath, clusterName, err)
 	}
 
-	cfg, err := knativetest.BuildClientConfig(configPath, clusterName)
+	kubeClient, err := kubernetes.NewForConfig(cfg)
```

Comment:
can't we do `c.KubeClient, err = kubernetes.NewForConfig(cfg)` ?

Reply:
we are setting it after checking the error